### ISSUE TITLE
List ConvertAll not supported.

### ISSUE THREAD

**Issue by michaelcheers** [bridgedotnet/Bridge#988]
The normal behavior is:

```c#
List<string> list = new List<string>{"1", "2", "3"};
List<int> converted = list.ConvertAll(int.Parse);
```
I have to now do:

```c#
List<string> list = new List<string>{"1", "2", "3"};
List<int> converted = new List<int>();
list.ForEach(v => converted.Add(int.Parse(v)));
```
instead.

I have a defenition for it:

```c#
public static List<T2> ConvertAll<T,T2> (this IEnumerable<T> value, Func<T, T2> function)
        {
            List<T2> result = new List<T2>();
            foreach (T item in value)
            {
                result.Add(function(item));
            }
            return result;
        }
```

**Comment by DaniilVeriga** [comment]
For references: `.ConvertAll()` [MSDN article[(https://msdn.microsoft.com/en-us/library/73fe8cwf%28v=vs.110%29.aspx).

@michaelcheers, just for your information - we are open to pull requests. More information in [CONTRIBUTING.md](https://github.com/bridgedotnet/Bridge/blob/master/.github/CONTRIBUTING.md).

**Comment by vladsch** [comment]
Your particular test case works fine
http://live.bridge.net/#79e83d05bf2733f4435b

But there are some limitations
See https://github.com/bridgedotnet/Bridge/issues/743

We are going to fix #743 in 1.12

**Comment by michaelcheers** [comment]
[Pull Request](https://github.com/bridgedotnet/Bridge/pull/997)
